File: ClearScript/V8/ClearScriptV8/V8TestProxyImpl.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.

#include "ClearScriptV8Managed.h"

namespace Microsoft {
namespace ClearScript {
namespace V8 {

    //-------------------------------------------------------------------------
    // V8TestProxyImpl implementation
    //-------------------------------------------------------------------------

    UIntPtr V8TestProxyImpl::GetNativeDigest(String^ gcValue)
    {
        StdString value(gcValue);
        return (UIntPtr)value.GetDigest();
    }

    //-------------------------------------------------------------------------

    V8TestProxy::Statistics^ V8TestProxyImpl::GetStatistics()
    {
        auto gcStatistics = gcnew Statistics;
        gcStatistics->IsolateCount = V8Isolate::GetInstanceCount();
        gcStatistics->ContextCount = V8Context::GetInstanceCount();
        return gcStatistics;
    }

    //-------------------------------------------------------------------------

    ENSURE_INTERNAL_CLASS(V8TestProxyImpl)

}}}

File: ClearScript/V8/SplitProxy/V8TestProxyImpl.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.

using System;

namespace Microsoft.ClearScript.V8.SplitProxy
{
    internal sealed class V8TestProxyImpl : V8TestProxy
    {
        public override UIntPtr GetNativeDigest(string value)
        {
            return V8SplitProxyNative.InvokeNoThrow(static (instance, value) => instance.V8UnitTestSupport_GetTextDigest(value), value);
        }

        public override Statistics GetStatistics()
        {
            var statistics = new Statistics();
            V8SplitProxyNative.InvokeNoThrow(static (instance, statistics) => instance.V8UnitTestSupport_GetStatistics(out statistics.IsolateCount, out statistics.ContextCount), statistics);
            return statistics;
        }

        public override void Dispose()
        {
        }
    }
}
